gui/default/syncthing/core/syncthingController.js

```diff
@@ -2096,12 +2096,19 @@ angular.module('syncthing.core')
             }
         };
 
-        $scope.otherDevices = function (devices) {
+        $scope.otherDevices = function(devices) {
             if (devices === undefined) {
                 devices = $scope.deviceList();
             }
-            return devices.filter(function (n) {
-                return n.deviceID !== $scope.myID;
+
+            var ids = devices.map(function (device) {
+                return device.deviceID
+            }).filter(function (id) {
+                return id !== $scope.myID
+            });
+
+            return $scope.deviceList().filter(function(n) {
+                return ids.includes(n.deviceID)
```

Comment:
```suggestion
                return ids.indexOf(n.deviceID) !== -1
```

For old browser compatibility.